Mathlib/CategoryTheory/Galois/Basic.lean

```diff
@@ -123,8 +127,77 @@ noncomputable instance : ReflectsColimitsOfShape (Discrete PEmpty.{1}) F :=
 noncomputable instance : PreservesFiniteLimits F :=
   preservesFiniteLimitsOfPreservesTerminalAndPullbacks F
 
+/-- Fibre functors reflect monomorphisms. -/
+instance foo : ReflectsMonomorphisms F := ReflectsMonomorphisms.mk <| by
+  intro X Y f _
+  haveI : IsIso (pullback.fst : pullback (F.map f) (F.map f) ⟶ F.obj X) :=
+    fst_iso_of_mono_eq (F.map f)
+  haveI : IsIso (F.map (pullback.fst : pullback f f ⟶ X)) := by
+    rw [← PreservesPullback.iso_hom_fst]
+    exact IsIso.comp_isIso
+  haveI : IsIso (pullback.fst : pullback f f ⟶ X) := isIso_of_reflects_iso pullback.fst F
+  exact (pullback.diagonal_isKernelPair f).mono_of_isIso_fst
+
+/-- Fibre functors are faithful. -/
+instance : Faithful F where
+  map_injective {X Y} f g h := by
+    haveI : IsIso (equalizer.ι (F.map f) (F.map g)) := equalizer.ι_of_eq h
+    haveI : IsIso (F.map (equalizer.ι f g)) := by
+      rw [← equalizerComparison_comp_π f g F]
+      exact IsIso.comp_isIso
+    haveI : IsIso (equalizer.ι f g) := isIso_of_reflects_iso _ F
+    exact eq_of_epi_equalizer
+
 end FibreFunctor
 
+variable (F : C ⥤ FintypeCat.{w}) [FibreFunctor F]
+
+private lemma FintypeCat.initial_iff_empty (X : FintypeCat.{w}) :
+    Nonempty (IsInitial X) ↔ IsEmpty X := by
+  rw [(IsInitial.isInitialIffObj FintypeCat.incl X).nonempty_congr, Types.initial_iff_empty]
+  rfl
+
+/-- An object is initial if and only if its fibre is empty. -/
+lemma initial_iff_fibre_empty (X : C) : Nonempty (IsInitial X) ↔ IsEmpty (F.obj X) := by
+  rw [(IsInitial.isInitialIffObj F X).nonempty_congr]
+  exact FintypeCat.initial_iff_empty (F.obj X)
+
+/-- An object is not initial, if its fibre is inhabited. -/
+lemma not_initial_of_inhabited {X : C} (x : F.obj X) (h : IsInitial X) : False :=
+  ((initial_iff_fibre_empty F X).mp ⟨h⟩).false x
```

Comment:
```suggestion
/-- An object whose fibre is inhabited is not initial. -/
lemma not_initial_of_inhabited {X : C} (x : F.obj X) : ¬(IsInitial X) :=
  ((initial_iff_fibre_empty F X).mp ⟨h⟩).false x
```
The statement is of course the same as your version, but I find a little inelegant to prove `False`.

Reply:
That does not work unfortunately, since `IsInitial X` is not `Prop`-valued.